sfge/src/main/java/com/salesforce/rules/ops/visitor/LoopDetectionVisitor.java

```diff
@@ -105,8 +146,27 @@ private void createOverridableLoopExclusion(
         }
     }
 
+    // necessary to limit the _afterLoopVisit method to only Method calls and Soql calls
     @Override
     public void afterVisit(MethodCallExpressionVertex vertex, SymbolProvider symbols) {
+        _afterLoopVisit(vertex, symbols);
+    }
+
+    @Override
+    public void afterVisit(SoqlExpressionVertex vertex, SymbolProvider symbols) {
+        _afterLoopVisit(vertex, symbols);
+    }
+
+    /**
+     * helper method to deal with removing the appropriate loop exclusion boundaries (see {@link
+     * #visit(MethodCallExpressionVertex, SymbolProvider)} and {@link #visit(SoqlExpressionVertex,
+     * SymbolProvider)})
+     */
+    private void _afterLoopVisit(BaseSFVertex vertex, SymbolProvider symbols) {
+        if (!(vertex instanceof SoqlExpressionVertex
+                || vertex instanceof MethodCallExpressionVertex)) {
+            afterVisit(vertex, symbols);
+        }
```

Comment:
If this method is only meant to be called with certain types of vertices, calling it with other types should throw a `ProgrammingException`.